src/handler/handler.c

```diff
@@ -0,0 +1,73 @@
+#include "handler/handler.h"
+
+#include <string.h>
+
+#include "hash_map.h"
+#include "server.h"
+#include "utility.h"
+
+// default callback function
+#include "handler/handlers/not_found.h"
+
+// static file handler
+#include "handler/handlers/static_handler.h"
+
+// handlers
+#include "handler/handlers/connect.h"
+#include "handler/handlers/dashboard.h"
+#include "handler/handlers/home.h"
+#include "handler/handlers/like.h"
+#include "handler/handlers/login.h"
+#include "handler/handlers/logout.h"
+#include "handler/handlers/post.h"
+#include "handler/handlers/profile.h"
+#include "handler/handlers/search.h"
+#include "handler/handlers/signup.h"
+#include "handler/handlers/unlike.h"
+
+static inline void add_handler(Hash_map* handlers, char* handling_name, Handler handler) {
```

Comment:
We should not add handlers here.
Each server has a set of handlers, and the facebooc server is a derived class of a server.
We could register some handlers into a base class, but not modify it directly.

Please refer to Open-closed principle.